keras/src/dtype_policies/dtype_policy_map.py

```diff
@@ -0,0 +1,105 @@
+import re
+from collections.abc import MutableMapping
+
+from keras.src import dtype_policies
+from keras.src.api_export import keras_export
+
+
+@keras_export(["keras.dtype_policies.DTypePolicyMap"])
+class DTypePolicyMap(MutableMapping):
```

Comment:
Would we want this to be a DTypePolicy subclass? Or have dtype policy accept this as a sub-component? Might be nice to keep everything that is passed as a dtype eventually resolving to a `DTypePolicy` object.